internal/core/src/simd/hook.cpp

```diff
@@ -118,52 +162,180 @@ find_term_hook() {
+    std::lock_guard<std::mutex> lock(hook_mutex);
+    std::string simd_type = "REF";
+#if defined(__x86_64__)
+    // Only support avx512 for now
+    if (use_avx512 && cpu_support_avx512()) {
+        simd_type = "AVX512";
+        equal_val_int8_t = EqualValAVX512<int8_t>;
+        equal_val_int16_t = EqualValAVX512<int16_t>;
+        equal_val_int32_t = EqualValAVX512<int32_t>;
+        equal_val_int64_t = EqualValAVX512<int64_t>;
+        equal_val_float = EqualValAVX512<float>;
+        equal_val_double = EqualValAVX512<double>;
+    }
+#endif
+    // TODO: support arm cpu
+    LOG_SEGCORE_INFO_ << "equal val hook simd type: " << simd_type;
+}
+
+void
+less_val_hook() {
+    static std::mutex hook_mutex;
+    std::lock_guard<std::mutex> lock(hook_mutex);
+    std::string simd_type = "REF";
+#if defined(__x86_64__)
+    // Only support avx512 for now
+    if (use_avx512 && cpu_support_avx512()) {
+        simd_type = "AVX512";
+        less_val_int8_t = LessValAVX512<int8_t>;
+        less_val_int16_t = LessValAVX512<int16_t>;
+        less_val_int32_t = LessValAVX512<int32_t>;
+        less_val_int64_t = LessValAVX512<int64_t>;
+        less_val_float = LessValAVX512<float>;
+        less_val_double = LessValAVX512<double>;
+    }
+#endif
+    // TODO: support arm cpu
+    LOG_SEGCORE_INFO_ << "less than val hook simd type: " << simd_type;
+}
+
+void
+greater_val_hook() {
+    static std::mutex hook_mutex;
+    std::lock_guard<std::mutex> lock(hook_mutex);
+    std::string simd_type = "REF";
+#if defined(__x86_64__)
+    // Only support avx512 for now
+    if (use_avx512 && cpu_support_avx512()) {
+        simd_type = "AVX512";
+        greater_val_int8_t = GreaterValAVX512<int8_t>;
+        greater_val_int16_t = GreaterValAVX512<int16_t>;
+        greater_val_int32_t = GreaterValAVX512<int32_t>;
+        greater_val_int64_t = GreaterValAVX512<int64_t>;
+        greater_val_float = GreaterValAVX512<float>;
+        greater_val_double = GreaterValAVX512<double>;
+    }
+#endif
+    // TODO: support arm cpu
+    LOG_SEGCORE_INFO_ << "greater than val hook simd type: " << simd_type;
+}
+
+void
+less_equal_val_hook() {
+    static std::mutex hook_mutex;
+    std::lock_guard<std::mutex> lock(hook_mutex);
+    std::string simd_type = "REF";
+#if defined(__x86_64__)
+    // Only support avx512 for now
+    if (use_avx512 && cpu_support_avx512()) {
+        simd_type = "AVX512";
+        less_equal_val_int8_t = LessEqualValAVX512<int8_t>;
+        less_equal_val_int16_t = LessEqualValAVX512<int16_t>;
+        less_equal_val_int32_t = LessEqualValAVX512<int32_t>;
+        less_equal_val_int64_t = LessEqualValAVX512<int64_t>;
+        less_equal_val_float = LessEqualValAVX512<float>;
+        less_equal_val_double = LessEqualValAVX512<double>;
+    }
+#endif
+    // TODO: support arm cpu
+    LOG_SEGCORE_INFO_ << "less equal than val hook simd type: " << simd_type;
+}
+
+void
+greater_equal_val_hook() {
+    static std::mutex hook_mutex;
```

Comment:
is possible use call_once?

Reply:
this interface will not be called by multi-thread for many times. Because init_hook_ interface which contains these interface init before main() and call_once is not necessary